### DISCUSSION TITLE
Median Spend and Median Balance

### DISCUSSION THREAD

**Discussion by aamiralia** [kinfoundation/kin-rfcs#119] [_ Archive - KRE 3.1 Discussion]
Just wanted to get some clarifications regarding the median balance and median spend. For the median spend calculations are you looking at all the individual spend transactions of at least 1000 Kin of the Active Users in the past 30 days? Or you are considering all users for this calculation? Same as with median balance, you are looking at the balance of the users who have at least spent 3 times in the past 30 days with each spend being at least 1000 Kin right?

Another question I have is regarding the caps. Are the caps applied to AUmax, BalanceMax and SpendMax? So for example the AUmax is the highest AU in the ecosystem or the highest AU cap = (AUapp / AUmax ) * AUBcapped in the ecosystem?

**Comment by kin1076** [comment]
This is a great question, if there are small spends in the app will this hurt the app by the KRE considering them in the median calculation?  If there were 500 spends of 1,000 Kin and 700 spends of 25 Kin is the median 1,000 or 25?

**Reply by skyf0xx** [reply]
> This is a great question, if there are small spends in the app will this hurt the app by the KRE considering them in the median calculation? If there were 500 spends of 1,000 Kin and 700 spends of 25 Kin is the median 1,000 or 25?

The 25 Kin spends would not be included in the calculation, so they wouldn't affect the app one way or another.

**Comment by skyf0xx** [comment]
For the Median spends and Median balances, we are getting them from the Active Users only. So for the calculation, we:
1. Collect the Active Users (3 times in 30 days, 1,000+ Kin)
2. Of these Active Users, get their:
   -  Median Balance
   - Median Spends

**Comment by skyf0xx** [comment]
AUmax is the highest AU in the ecosystem.

If a smaller app had 1,000 AU, and the largest app had 10,000 AU, the smaller app's AU cap would be:

AU cap = (1,000 / 10,000 ) * AUBcapped

This translates as apps having a larger AU finding themselves less capped (or rewarded more) due to their AU contribution.

This is similar to apps with higher spends and higher median balances.

**Comment by showered** [comment]
@hitwill
Are you sure you meant to write "median" in the proposal, or should it have been "mean"?

The illustrative figures shown in the table seem far too precise to be median.

**Reply by skyf0xx** [reply]
> @hitwill Are you sure you meant to write "median" in the proposal, or should it have been "mean"?
> 
> The illustrative figures shown in the table seem far too precise to be median.

It should be Median. We considered Mean but that's easier to game by having one large player vs many players who fit the Median. @showered Could you be more specific about the table?

**Reply by showered** [reply]
The [top 3 apps' Median Spends](https://github.com/kinecosystem/rewards-engine/blob/master/improvement-proposals/kf_kre_3_1.md#impact-of-transitioning-to-kre-31) are shown as 24,719, 3,748, 5,745.

The nature of Median average suggests that the **_typical_** user of these apps is spending the above amounts. The amounts provided are very specific numbers that would surely only ever be spent by 1 or 2 users at most (possibly via a "withdrawal"). With Median I would have expected the average spend to be a round number like 10, 100, 500.

**Reply by kin1076** [reply]
Median isn't an average (you know that but just wanted to clarify for Will that you're not saying it is) but it would suggest theoretically that the typical person is spending that amount. It DOES look like an average (mean). Trying to think of what those spends could have been...

**Reply by skyf0xx** [reply]
I see what you mean. Some apps do have different denominations based on what a user wants to spend. (If they allow the user to enter the value)

For example, [this tx is from an app with a 3,748 spend](https://explorer.solana.com/tx/32h1b2ER6u5pzokyRrjtYTxMtNtQ86FFEaLXEQE8fsKhuecvsu94Pj5AL6gzyeYXroeNuXgakBj8KB84PzEmuW8V) and [this one from a 5,745 spend](https://explorer.solana.com/tx/2kbZ1tcvgTaVMR3wUXaDxWeHYfnTEJVAUhuMitD3kDczhXsYPYadnQjgYfHd9t96jX88ZTLxNLYPHvwAXvs2RALe)

The median just orders the different spends an app has from smallest to greatest and picks the one in the middle. It prevents gaming by creating a few massive spends or balances in order to raise the average.

**Reply by showered** [reply]
Thanks for getting back to me @hitwill 

The second transaction you linked to is actually an Earn, not a Spend. Also one is from January 2021 (13 months ago) and the other one is October 2021.

**Reply by showered** [reply]
Further to my previous comment, I think what has probably happened is that one of the apps is using a wallet as an Earn wallet that your KRE calculator isn't aware of. Therefore all the Earns being sent from [that wallet](https://explorer.solana.com/address/HsXqrSYPCMEjttqsQYdWLgoXeVdCWfRaEJ9YxE7c7RWp) are being counted as Spends, and therefore they contribute to and skew the Median spend figure.

**Reply by skyf0xx** [reply]
@showered apologies - I just meant to show there are plenty of transactions with non rounded amount. [This one](https://explorer.solana.com/tx/HdyoQsJBPYBAmCfANJgUpNyri2Y3LxNwdovVrQCGT9A4XXM95R39NpmMnPgztwHPUHQSBWwGYVwpbZ45hanrRaG) should be a P2P in November

**Reply by skyf0xx** [reply]
> Further to my previous comment, I think what has probably happened is that one of the apps is using a wallet as an Earn wallet that your KRE calculator isn't aware of. Therefore all the Earns being sent from [that wallet](https://explorer.solana.com/address/HsXqrSYPCMEjttqsQYdWLgoXeVdCWfRaEJ9YxE7c7RWp) are being counted as Spends, and therefore they contribute to and skew the Median spend figure.

That could be a possibility. A few spends would still be negligible since we're looking at Median. 

However, many high spends could skew it more. I have added it as something we need to check for degree of skew. One option is to average the spends from each single account, then take a median of that. That would remove the effect of a single account having many large spends and skewing the median.
